test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelpersTest.cs

```diff
@@ -148,5 +148,110 @@ public class TagHelpersTests
             var responseContent = await response.Content.ReadAsStringAsync();
             Assert.Equal(expectedContent, responseContent);
         }
+
+        [Fact]
+        public async Task CacheTagHelper_CanCachePortionsOfViewsPartialViewsAndViewComponents()
```

Comment:
this functional test is for custom tag helpers, not the built-in MVC ones.  either hold off on the functional test 'til @ianhong pushes #1770 or rebase on his branch.